rust/src/editor.rs

```diff
@@ -198,7 +191,26 @@ impl Editor {
         }
     }
 
+    fn delete_forward(&mut self) {
+        if self.view.sel_start == self.view.sel_end {
+            self.move_right();
+        }
+
+        self.delete();
+    }
+
     fn delete_backward(&mut self) {
+        self.delete();
+    }
+
+    fn delete_to_beginning_of_line(&mut self) {
+        self.modify_selection();
+        self.move_to_left_end_of_line();
+
+        self.delete();
```

Comment:
This has interesting behavior when the cursor is already at the beginning of the line - it deletes the newline at the end of the previous line. That could actually be useful behavior but doesn't match the Cocoa model (TextEdit does no-op in that case).